source/image/ImageBuffer.cpp

```diff
@@ -359,6 +379,136 @@ namespace {
 
 
 
+	// Read an AVIF file, and return the number of frames. This might be
+	// greater than the number of frames in the file due to frame time corrections.
+	// Since sprite animation properties are not visible here, we take the shortest frame
+	// duration, and treat that as our time unit. Every other frame is repeated
+	// based on how much longer its duration is compared to this unit.
+	// TODO: If animation properties are exposed here, we can have custom presentation
+	// logic that avoids duplicating the frames.
+	int ReadAVIF(const filesystem::path &path, ImageBuffer &buffer, int frame, bool alphaPreMultiplied)
+	{
+		unique_ptr<avifDecoder, void(*)(avifDecoder *)> decoder(avifDecoderCreate(), avifDecoderDestroy);
+		if(!decoder)
+		{
+			Logger::LogError("Could not create avif decoder");
+			return 0;
+		}
+		// Maintenance note: this is where decoder defaults should be overwritten (codec, exif/xmp, etc.)
+
+		string data = Files::Read(path);
+		avifResult result = avifDecoderSetIOMemory(decoder.get(), reinterpret_cast<const uint8_t *>(data.c_str()),
+				data.size());
+		if(result != AVIF_RESULT_OK)
+		{
+			Logger::LogError("Could not read file: " + path.generic_string());
+			return 0;
+		}
+
+		result = avifDecoderParse(decoder.get());
+		if(result != AVIF_RESULT_OK)
+		{
+			Logger::LogError("Failed to decode image: " + string(avifResultToString(result)));
+			return 0;
+		}
+		// Generic image information is now available (width, height, depth, color profile, metadata, alpha, etc.)
+		// Also image count and frame timings.
+		if(!decoder->imageCount)
+			return 0;
+
+		// Find the shortest frame duration.
+		double frameTimeUnit = -1;
+		avifImageTiming timing;
+		for(int i = 0; i < decoder->imageCount; ++i)
+		{
+			result = avifDecoderNthImageTiming(decoder.get(), 0, &timing);
```

Comment:
Isn't this supposed to be the frame index?
```suggestion
			result = avifDecoderNthImageTiming(decoder.get(), i, &timing);
```